src/utils/clone.ts

```diff
@@ -0,0 +1,31 @@
+export function cloneShadowRoots(sourceNode: Node, targetNode: Node) {
+    if(sourceNode instanceof Element && targetNode instanceof Element) {
+        const shadowRoot = sourceNode.shadowRoot;
+        if (shadowRoot !== null) {
+            // Create a new Shadow Root for the target element
+            const copiedShadowRoot = targetNode.attachShadow({mode: shadowRoot.mode});
+
+            // Clone the contents of the original ShadowRoot manually
+            const originalShadowContent = shadowRoot.innerHTML;
+
+            // Create a temporary container to hold the content to copy
+            const tempContainer = targetNode.ownerDocument.createElement(shadowRoot.children[0].tagName);
+            tempContainer.innerHTML = originalShadowContent;
+
+            // Clone and append the Shadow DOM content
+            copiedShadowRoot.appendChild(tempContainer);
+
+            // Recursively clone any nested Shadow DOMs within this Shadow DOM content
```

Comment:
This recursion and the one below is what scared me from doing this in the past, I'm worried about performance problems on pages with large amounts of shadow content. Maybe we worry about that if/when it happens though

Reply:
What do think about a warning in the README to address this issue.
I optimized/simplified the clone routine, treewalker should be quite efficient. Please have a look.

Reply:
I like that, thanks!